src/main/java/seedu/toluist/dispatcher/CommandDispatcher.java

```diff
@@ -43,21 +45,41 @@
     private final EventsCenter eventsCenter = EventsCenter.getInstance();
     private final AliasTable aliasConfig = Config.getInstance().getAliasTable();
 
+    /**
+     * ArrayList to store previous commands entered since starting the application
+     */
+    private ArrayList<String> commandHistory;
+    private Integer historyPointer = 0;
```

Comment:
hmm `int` in java is a literal type. `Integer` is only used if you wanna have sth like `List<Integer>` coz `List<int>` is not legal